SingularityService/src/main/java/com/hubspot/singularity/config/TaskQuickLink.java

```diff
@@ -0,0 +1,35 @@
+package com.hubspot.singularity.config;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * TaskQuickLink defines a "quick-link" button to show on the
+ */
+public class TaskQuickLink {
+  /**
+   * Name of the button label in the SingularityUI interface
+   */
+  @JsonProperty
+  private final String name;
+  /**
+   * Handlebars string to parse to generate a URL for the quick link
+   * (e.g. "http://{{TASK_HOST}}/{{TASK_REQUEST_ID}}")
+   */
+  @JsonProperty
+  private final String url;
+
+  @JsonCreator
+  public TaskQuickLink(@JsonProperty("name") String name, @JsonProperty("url") String url) {
```

Comment:
I think a second PR makes sense, it doesn't change the structure of how the TaskQuickLinks are specified, just an optional additional field if the quick link isn't supposed to apply to every type.